disperser/batcher/finalizer.go

```diff
@@ -72,55 +86,75 @@ func (f *finalizer) FinalizeBlobs(ctx context.Context) error {
 	if err != nil {
 		return fmt.Errorf("FinalizeBlobs: error getting latest finalized block: %w", err)
 	}
+	lastFinalBlock := finalizedHeader.Number.Uint64()
 
-	metadatas, err := f.blobStore.GetBlobMetadataByStatus(ctx, disperser.Confirmed)
-	if err != nil {
-		return fmt.Errorf("FinalizeBlobs: error getting blob headers: %w", err)
-	}
-
-	f.logger.Info("FinalizeBlobs: finalizing blobs", "numBlobs", len(metadatas), "finalizedBlockNumber", finalizedHeader.Number)
-
-	for _, m := range metadatas {
-		blobKey := m.GetBlobKey()
-		confirmationMetadata, err := f.blobStore.GetBlobMetadata(ctx, blobKey)
+	totalProcessed := 0
+	totalFinalized := 0
+	totalFailed := 0
+	metadatas, exclusiveStartKey, err := f.blobStore.GetBlobMetadataByStatusWithPagination(ctx, disperser.Confirmed, f.numBlobsPerFetch, nil)
+	for len(metadatas) > 0 {
```

Comment:
Think it would be good to add a metric for how long this part of the system takes. Don't think we're measuring it right now.

Reply:
good call. added